src/System.IO.FileSystem/src/System/IO/Win32FileStream.cs

```diff
@@ -1684,6 +1685,299 @@ private int GetLastWin32ErrorAndDisposeHandleIfInvalid(bool throwIfInvalidHandle
             return errorCode;
         }
 
+        public override Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken)
+        {
+            // Validate arguments as would the base implementation
+            if (destination == null) throw new ArgumentNullException(nameof(destination));
+            if (bufferSize <= 0) throw new ArgumentOutOfRangeException(nameof(bufferSize), SR.ArgumentOutOfRange_NeedPosNum);
+            bool parentCanRead = _parent.CanRead;
+            if (!parentCanRead && !_parent.CanWrite) throw new ObjectDisposedException(null, SR.ObjectDisposed_StreamClosed);
+            bool destinationCanWrite = destination.CanWrite;
+            if (!destination.CanRead && !destinationCanWrite) throw new ObjectDisposedException(nameof(destination), SR.ObjectDisposed_StreamClosed);
+            if (!parentCanRead) throw new NotSupportedException(SR.NotSupported_UnreadableStream);
+            if (!destinationCanWrite) throw new NotSupportedException(SR.NotSupported_UnwritableStream);
+            if (_handle.IsClosed) throw Error.GetFileNotOpen();
```

Comment:
Shouldn't this go after `Task.FromCanceled` to match previous behavior?  Also seems more correct to me, if app has requested cancellation it could be because they know the stream is closed.